xtc/src/ledger.rs

```diff
@@ -97,11 +181,38 @@ static InsufficientBalanceError: ErrorDetails = ErrorDetails {
     msg: "Insufficient Balance",
     code: APIError::InsufficientBalance,
 };
+
+static InsufficientAllowanceError: ErrorDetails = ErrorDetails {
+    msg: "Insufficient Allowance",
+    code: APIError::InsufficientAllowance,
+};
+
 static UnknownError: ErrorDetails = ErrorDetails {
     msg: "Unknown",
     code: APIError::Unknown,
 };
 
+// Disabled as the `name` clashes with a similar method in the cycles wallet
```

Comment:
Is this something we just have to do without?

Reply:
I'd also say ERC-20's name should have higher priority. So let's uncomment this and comment out the cycles_wallet's name method.